recipe_scrapers/monsieurcuisine.py

```diff
@@ -1,72 +1,119 @@
 # mypy: disallow_untyped_defs=False
-from ._abstract import AbstractScraper
-from ._utils import get_minutes, get_yields, normalize_string
+import re
+
+import requests
+
+from ._abstract import HEADERS, AbstractScraper
+from ._utils import get_yields
+
+SCRIPT_PATTERN = re.compile(r'"recipeId":(\d+)')
 
 
 class MonsieurCuisine(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+        scripts = self.soup.find_all("script")
+        recipe_id = None
+        for script in scripts:
+            matches = SCRIPT_PATTERN.search(str(script.string))
+            if matches:
+                recipe_id = matches.group(1)
+        language_iso = self.soup.find("html")["lang"]
+        data_url = f"https://mc-api.tecpal.com/api/v2/recipes/{recipe_id}"
+        headers = {"Accept-Language": language_iso, "Device-Type": "web"}
+        headers.update(HEADERS)
```

Comment:
Nitpick: let's flip these around, using `HEADERS` as the default, and then adding our modifications:

```suggestion
        headers = HEADERS.copy()
        headers.update({"Accept-Language": language_iso, "Device-Type": "web"})
```

(the `.copy()` is important, because otherwise we'd be changing the settings for subsequent calls)